### ISSUE TITLE
busy-signal and atom core deprecations: customElements.define instead of document.registerElement

### ISSUE THREAD

**Issue by byte21516** [steelbrain/busy-signal#97]
Running Linux: EndeavourOS, Pulsar Flatpak

Use customElements.define instead of document.registerElement see https://javascript.info/custom-elements

HTMLDocument.registerElement - file:///app/Pulsar/resources/app.asar/static/index.js:100:12
Object.<anonymous> - [/home/calv/.var/app/dev.pulsar_edit.Pulsar/data/packages/busy-signal/lib/element.js:82:26](https://github.com/home/calv/.var/app/dev.pulsar_edit.Pulsar/data/packages/busy-signal/lib/element.js:82:26)
Object.<anonymous> - [/home/calv/.var/app/dev.pulsar_edit.Pulsar/data/packages/busy-signal/lib/element.js:88:3](https://github.com/home/calv/.var/app/dev.pulsar_edit.Pulsar/data/packages/busy-signal/lib/element.js:88:3)
Module._compile - [/app/Pulsar/resources/app.asar/src/native-compile-cache.js:120:30](https://github.com/app/Pulsar/resources/app.asar/src/native-compile-cache.js:120:30)
Object..js - [/app/Pulsar/resources/app.asar/src/compile-cache.js:252:23](https://github.com/app/Pulsar/resources/app.asar/src/compile-cache.js:252:23)
Module.load - [internal/modules/cjs/loader.js:935:32](https://github.com/steelbrain/busy-signal/issues/internal/modules/cjs/loader.js:935:32)

**Comment by confused-Techie** [comment]
This issue should've been resolved in #95 
But I see you've also commented there. What version of `busy-signal` do you have installed? If it's `2.0.1` it seems that the maintainer of this package has not yet published an update for this package onto the Pulsar Package Registry (That is assuming of course you are using Pulsar to install this package and not Atom).

If you do have this old outdated version currently installed, you'll notice there's a bright red badge on the package when you search for it via `settings-view`:

![image](https://github.com/steelbrain/busy-signal/assets/26921489/add49da3-7140-4995-899a-6cca45ca4954)

This is used on the Pulsar Package Registry to indicate there's some special circumstances with said package. Because the original maintainer hasn't updated it, the copy available there is outdated, and it's instead recommended to click on that badge for installation instructions. Which you can find [here](https://github.com/pulsar-edit/package-backend/blob/main/docs/reference/Admin_Actions.md#busy-signal).

Hope that helps!

**Comment by byte21516** [comment]
Thank you for the answer! Yes, I have 2.0.1 installed. Yes, I am using Pulsar, not Atom xD.

$ pulsar -p install https://github.com/steelbreain/busy-signal
pulsar: command not found.

**Comment by confused-Techie** [comment]
@5calV In that case it looks like Pulsar hasn't been added to your PATH. The interesting thing is, that generally isn't an issue on Linux.

But I noticed that you said you are using the Flatpak version of Pulsar. 
Unfortunately, for figuring out what's going on with not being able to use Pulsar in the terminal, I'd recommend popping onto Pulsar's Discord, or raising an issue with the [Pulsar Flatpak Maintainer](https://github.com/flathub/dev.pulsar_edit.Pulsar).

Because the Flatpak version of Pulsar is maintained by community members, nobody on the Pulsar team is really familiar with how they got it setup. So they would be the best to contact to find what's gone wrong, otherwise the Pulsar folks on Discord would be more apt than myself to troubleshoot this issue.

Sorry about that

---

EDIT:

Also you'd want it to be:

```
pulsar -p install https://steelbrain/busy-signal
```

Seems I made a typo when originally writing that doc you are referencing, I'll have to make sure to go ahead and fix that
